Lib/test/test_complex.py

```diff
@@ -632,6 +632,22 @@ def test_constructor_negative_nans_from_string(self):
         self.assertEqual(copysign(1., complex("-nan-nanj").real), -1.)
         self.assertEqual(copysign(1., complex("-nan-nanj").imag), -1.)
 
+    def test_other_non_numeric_input_types(self):
+        class CustomStr(str): ...
+        class CustomBytes(bytes): ...
+        class CustomByteArray(bytearray): ...
+
+        factories = [bytes, bytearray, lambda b: CustomStr(b.decode()),
+                     CustomBytes, CustomByteArray]
+
+        for f in factories:
+            x = f(b"(1+1j)")
```

Comment:
this is a part of `subTest`

Reply:
Now factory value is used as a parameter.